firedrake/mesh.py

```diff
@@ -894,34 +909,55 @@ def cell_to_facets(self):
                        name="cell-to-local-facet-dat")
 
     def num_cells(self):
-        cStart, cEnd = self._topology_dm.getHeightStratum(0)
+        cStart, cEnd = self.topology_dm.getHeightStratum(0)
         return cEnd - cStart
 
     def num_facets(self):
-        fStart, fEnd = self._topology_dm.getHeightStratum(1)
+        fStart, fEnd = self.topology_dm.getHeightStratum(1)
         return fEnd - fStart
 
     def num_faces(self):
-        fStart, fEnd = self._topology_dm.getDepthStratum(2)
+        fStart, fEnd = self.topology_dm.getDepthStratum(2)
         return fEnd - fStart
 
     def num_edges(self):
-        eStart, eEnd = self._topology_dm.getDepthStratum(1)
+        eStart, eEnd = self.topology_dm.getDepthStratum(1)
         return eEnd - eStart
 
     def num_vertices(self):
-        vStart, vEnd = self._topology_dm.getDepthStratum(0)
+        vStart, vEnd = self.topology_dm.getDepthStratum(0)
         return vEnd - vStart
 
     def num_entities(self, d):
-        eStart, eEnd = self._topology_dm.getDepthStratum(d)
+        eStart, eEnd = self.topology_dm.getDepthStratum(d)
         return eEnd - eStart
 
     @utils.cached_property
     def cell_set(self):
         size = list(self._entity_classes[self.cell_dimension(), :])
         return op2.Set(size, "Cells", comm=self.comm)
 
+    def set_partitioner(self, distribute):
```

Comment:
A few people have been asking to be able to configure the partitioner. This also needs a docstring.

Reply:
Is there a way to configure partitioners via PETSc? Maybe I misunderstood your comment.

Reply:
So we can run `-petscpartitioner_type foo` on the commandline, but that's not very discoverable. I guess here `set_partitioner` seems to imply that we can select it, but really we're just setting it up.

Maybe `set_partitioner(self, distribute, partitioner_type=None)` or something. Or we can save that for a separate round.

Reply:
That makes sense. 
Created a separate PR just for things around partitioner (https://github.com/firedrakeproject/firedrake/pull/1934).